src/hdmf/common/resources.py

```diff
@@ -408,7 +409,29 @@ def _get_file_from_container(self, **kwargs):
                 msg = 'Could not find file. Add container to the file.'
                 raise ValueError(msg)
 
-    @docval({'name': 'root_container',  'type': HERDManager,
+    @docval({'name': 'objects', 'type': list,
+             'doc': 'List of objects to check for TermSetWrapper within the fields.'})
+    def __check_termset_wrapper(self, **kwargs):
+        """
+        Takes a list of objects and checks the fields for TermSetWrapper.
+        :return: [[object1, attribute_name1, wrapper1], [object2, attribute_name2, wrapper2], ...]
+        """
+        objects = kwargs['objects']
+
+        ret = [] # list to be returned with the objects, attributes and corresponding termsets
+
+        for obj in objects:
+            # Get all the fields, parse out the methods and internal variables
+            obj_fields = [a for a in dir(obj) if not a.startswith('_') and not callable(getattr(obj, a))]
```

Comment:
This coarse approach still unsettles me, but I cannot think of a specific real-world example of a problem that this might cause.

Reply:
I can open an enhancement ticket for us to look at down the road?

Reply:
Instead of using `dir` to determine `obj_fields` I think you should be able to just use``__dict__`` https://docs.python.org/3/library/stdtypes.html#object.__dict__  :

```
for attribute_name, attribute_value in obj.__dict__.items():
    if isinstance(attribute_value, TermSetWrapper):
          ret.append([obj, attribute_name, attribute_value])
```

Reply:
It's either dir(obj) or obj.__dict__.keys(). Either way I have to still clean out the "_". I don't think there's a tangible benefit on switching.